src/main/java/io/vertx/ext/asyncsql/impl/pool/AsyncConnectionPool.java

```diff
@@ -41,25 +43,29 @@
   public static final int DEFAULT_MAX_POOL_SIZE = 10;
   public static final int DEFAULT_MAX_CONNECTION_RETRIES = 0;       // No connection retries by default
   public static final int DEFAULT_CONNECTION_RETRY_DELAY = 5_000;   // 5 seconds between retries by default
+  public static final int DEFAULT_CONNECTION_RELEASE_DELAY = 60_000;// 1 minute until release idle connection
 
   private static final Logger logger = LoggerFactory.getLogger(AsyncConnectionPool.class);
 
   private final int maxPoolSize;
   private final int maxConnectionRetries;
   private final int connectionRetryDelay;
+  private final int connectionReleaseDelay;
 
   protected final Configuration connectionConfig;
   protected final Vertx vertx;
 
   private int poolSize = 0;
   private final Deque<Connection> availableConnections = new ArrayDeque<>();
   private final Deque<Handler<AsyncResult<Connection>>> waiters = new ArrayDeque<>();
+  private final Map<Connection,Long> timers = new HashMap<>();
 
   public AsyncConnectionPool(Vertx vertx, JsonObject globalConfig, Configuration connectionConfig) {
     this.vertx = vertx;
     this.maxPoolSize = globalConfig.getInteger("maxPoolSize", DEFAULT_MAX_POOL_SIZE);
     this.maxConnectionRetries = globalConfig.getInteger("maxConnectionRetries", DEFAULT_MAX_CONNECTION_RETRIES);
     this.connectionRetryDelay = globalConfig.getInteger("connectionRetryDelay", DEFAULT_CONNECTION_RETRY_DELAY);
+    this.connectionReleaseDelay = globalConfig.getInteger("connectionReleaseDelay", DEFAULT_CONNECTION_RELEASE_DELAY);
```

Comment:
As you have `0` as another special value - why not use that in order to not break code relying on over 1 minute?